src/gtk/dataobj.cpp

```diff
@@ -110,10 +110,48 @@ wxDataFormatId wxDataFormat::GetType() const
 
 wxString wxDataFormat::GetId() const
 {
+#ifdef __WXGTK4__
+    return wxString::FromAscii(m_format);
+#else
     wxGtkString atom_name(gdk_atom_name(m_format));
     return wxString::FromAscii(atom_name);
+#endif
 }
 
+#ifdef __WXGTK4__
+void wxDataFormat::SetId( NativeFormat format )
+{
+    PrepareFormats();
+    m_format = g_intern_string(format);
```

Comment:
It would be better to `#if/#else` this one line rather than duplicate the entire function.